kappa/event_source/cloudwatch.py

```diff
@@ -56,13 +56,22 @@ def add(self, function):
             response = self._events.call('put_rule', **kwargs)
             LOG.debug(response)
             self._config['arn'] = response['RuleArn']
-            response = self._lambda.call('add_permission',
-                                         FunctionName=function.name,
-                                         StatementId=str(uuid.uuid4()),
-                                         Action='lambda:InvokeFunction',
-                                         Principal='events.amazonaws.com',
-                                         SourceArn=response['RuleArn'])
-            LOG.debug(response)
+            existingPermission={}
+            try:
+                response = self._lambda.call('get_policy',
+                                         FunctionName=function.name)
+                existingPermission = self._config['arn'] in str(response['Policy'])
```

Comment:
Can you expand on that?

All this section does is add a permission to the policy that allows the rule to invoke the function. 

If the rule arn already exists in the policy, then we determine that the permission has already been added and there is no need to add it again. This still allows for permissions to be added for other rules.